worlds/sc2/MissionTables.py

```diff
@@ -138,7 +138,7 @@ def _asdict(self):
 
 class MissionInfo(NamedTuple):
     mission: SC2Mission
-    required_world: List[MissionConnection]
+    required_world: List[Union[MissionConnection, Dict[Literal["campaign", "connect_to"], int]]]
```

Comment:
This was probably the tricksiest update to verify. Within MissionTables.py, `required_world` is always set to a `MissionConnection`. In the client, however, the contents of this variable are always parsed to `parse_unlock`, which distinctly does not take a `MissionConnection`. Debugging revealed the dictionary contents was what always populated the variable; `parse_unlock` can technically also handle a basic `int` for legacy reasons but I don't see the point of annotating that here.

Reply:
The thing is the lifecycle, as generally classes (except core) can't live within the server, therefore the serializing to dict and back. The int variant is legacy data so it's transformed to global connection